src/Umbraco.Infrastructure/Migrations/Upgrade/UmbracoPlan.cs

```diff
@@ -1,289 +1,56 @@
-            To<NoopMigration>("{03482BB0-CF13-475C-845E-ECB8319DBE3C}");
-
-            // This should be safe to execute again. We need it with a new name to ensure updates from all the following has executed this step.
-            // - 8.15.0 RC    - Current state: {4695D0C9-0729-4976-985B-048D503665D8}
-            // - 8.15.0 Final - Current state: {5C424554-A32D-4852-8ED1-A13508187901}
-            // - 9.0.0 RC1    - Current state: {5060F3D2-88BE-4D30-8755-CF51F28EAD12}
-            To<UpdateCmsPropertyGroupIdSeed>("{622E5172-42E1-4662-AD80-9504AF5A4E53}");
-            To<ExternalLoginTableIndexesFixup>("{10F7BB61-C550-426B-830B-7F954F689CDF}");
-            To<DictionaryTablesIndexes>("{5AAE6276-80DB-4ACF-B845-199BC6C37538}");
-
-            // to 9.0.0 RC1
-            To<MigrateLogViewerQueriesFromFileToDb>("{22D801BA-A1FF-4539-BFCC-2139B55594F8}");
-            To<ExternalLoginTableIndexes>("{50A43237-A6F4-49E2-A7A6-5DAD65C84669}");
-            To<ExternalLoginTokenTable>("{3D8DADEF-0FDA-4377-A5F0-B52C2110E8F2}");
-            To<MemberTableColumns>("{1303BDCF-2295-4645-9526-2F32E8B35ABD}");
-            To<AddPasswordConfigToMemberTable>("{5060F3D2-88BE-4D30-8755-CF51F28EAD12}");
-            To<AddPropertyTypeGroupColumns>(
-                "{A2686B49-A082-4B22-97FD-AAB154D46A57}"); // Re-run this migration to make sure it has executed to account for migrations going out of sync between versions.
-
-            // TO 9.0.0-rc4
-            To<UmbracoServerColumn>(
-                "5E02F241-5253-403D-B5D3-7DB00157E20F"); // Jaddie: This GUID is missing the { }, although this likely can't be changed now as it will break installs going forwards
-
-            // TO 9.1.0
-            To<AddContentVersionCleanupFeature>("{8BAF5E6C-DCB7-41AE-824F-4215AE4F1F98}");
-
-            // TO 9.2.0
-            To<AddUserGroup2NodeTable>("{0571C395-8F0B-44E9-8E3F-47BDD08D817B}");
-            To<AddDefaultForNotificationsToggle>("{AD3D3B7F-8E74-45A4-85DB-7FFAD57F9243}");
-
-
-
-            // TO 9.3.0
-            To<MovePackageXMLToDb>("{A2F22F17-5870-4179-8A8D-2362AA4A0A5F}");
-            To<UpdateExternalLoginToUseKeyInsteadOfId>("{CA7A1D9D-C9D4-4914-BC0A-459E7B9C3C8C}");
-            To<AddTwoFactorLoginTable>("{0828F206-DCF7-4F73-ABBB-6792275532EB}");
-
-            // TO 9.4.0
-            To<AddScheduledPublishingLock>("{DBBA1EA0-25A1-4863-90FB-5D306FB6F1E1}");
-            To<UpdateRelationTypesToHandleDependencies>("{DED98755-4059-41BB-ADBD-3FEAB12D1D7B}");
+            // .From("state-1")
+            // .To<ChangeA>("state-a")
+            // .To<ChangeB>("state-b") // Some might already be in this state, without having applied ChangeA
+            //
+            // .From("state-1")
+            // .Merge()
+            //     .To<ChangeA>("state-a")
+            // .With()
+            //     .To<ChangeB>("state-b")
+            // .As("state-2");
+
+            From(InitialState)
+                .To("{init-10.0.0}");
```

Comment:
This is currently not doing anything (besides blocking you to use a v10 upgraded database with v9 again without manually resetting the migration state in the database), but not specifying a plan throws an exception during migration and we will probably have some migrations before the final release.

The only major task that still needs doing is to ensure the `DatabaseSchemaCreator` and `DatabaseDataCreator` also include all the changes done by the migrations removed above! ⚠️